source/core/src/main/com/csse3200/game/components/npc/attack/BossRangeAttackComponent.java

```diff
@@ -100,6 +104,9 @@ private void shoot() {
 
     private void singleShoot(Vector2 spawnLocation, Vector2 movingDirection) {
         Entity projectile = projectileFactory.create(getProjectileNames()[getAnimationID()], movingDirection, spawnLocation);
+        ProjectileAttackComponent projectileAttackComponent = projectile.getComponent(ProjectileAttackComponent.class);
+        Vector2 v = projectileAttackComponent.getSpeed();
+        projectileAttackComponent.setSpeed(new Vector2(v.x * speedCoefficient, v.y * speedCoefficient));
```

Comment:
can use Vector.scl which should also work but no issues with this

Reply:
So you good to approve it?